File: 1-easy/problems/3083-existence-of-a-substring-in-a-string-and-its-reverse/solution.ts
function isSubstringPresent(s: any): boolean | number | string | any {
  """
  :type s: str
  :rtype: bool
  """
  lookup = [[false]*26 for _ in Array.from({length: 26}, (_, i) => i)]
  for (i in Array.from({length: (s}, (_, i) => i).length-1)) {
      lookup[ord(s[i])-ord('a')][ord(s[i+1])-ord('a')] = true
  return any(lookup[ord(s[i+1])-ord('a')][ord(s[i])-ord('a')]  for i in Array.from({length: (s}, (_, i) => i).length-1));
}

export { isSubstringPresent };

File: 1-easy/problems/3083-existence-of-a-substring-in-a-string-and-its-reverse/solution.py
class Solution:
    def isSubstringPresent(self, s):
        """
        :type s: str
        :rtype: bool
        """
        lookup = [[False]*26 for _ in range(26)]
        for i in range(len(s)-1):
            lookup[ord(s[i])-ord('a')][ord(s[i+1])-ord('a')] = True
        return any(lookup[ord(s[i+1])-ord('a')][ord(s[i])-ord('a')]  for i in range(len(s)-1))
    

import collections


# hash table
class Solution2(object):
    def isSubstringPresent(self, s):
        """
        :type s: str
        :rtype: bool
        """
        lookup = collections.defaultdict(set)
        for i in range(len(s)-1):
            lookup[s[i]].add(s[i+1])
        return any(s[i] in lookup[s[i+1]] for i in range(len(s)-1))
